src/qt/optionsmodel.cpp

```diff
@@ -606,4 +625,41 @@ void OptionsModel::checkAndMigrate()
     if (settings.contains("addrSeparateProxyTor") && settings.value("addrSeparateProxyTor").toString().endsWith("%2")) {
         settings.setValue("addrSeparateProxyTor", GetDefaultProxyAddress());
     }
+
+    // Migrate and delete legacy GUI settings that have now moved to <datadir>/settings.json.
+    auto migrate_setting = [&](OptionID option, const QString& qt_name, const std::string& name) {
+        if (!settings.contains(qt_name)) return;
+        QVariant value = settings.value(qt_name);
+        if (node().getPersistentSetting(name).isNull()) {
+            if (option == ProxyIP) {
+                ProxySetting parsed = ParseProxyString(value.toString());
+                setOption(ProxyIP, parsed.ip);
+                setOption(ProxyPort, parsed.port);
+            } else if (option == ProxyIPTor) {
+                ProxySetting parsed = ParseProxyString(value.toString());
+                setOption(ProxyIPTor, parsed.ip);
+                setOption(ProxyPortTor, parsed.port);
+            } else {
+                setOption(option, value);
+            }
+        }
+        settings.remove(qt_name);
+    };
+
+    migrate_setting(DatabaseCache, "nDatabaseCache", "dbcache");
+    migrate_setting(ThreadsScriptVerif, "nThreadsScriptVerif", "par");
+#ifdef ENABLE_WALLET
+    migrate_setting(SpendZeroConfChange, "bSpendZeroConfChange", "spendzeroconfchange");
+    migrate_setting(ExternalSignerPath, "external_signer_path", "signer");
+#endif
+    migrate_setting(MapPortUPnP, "fUseUPnP", "upnp");
+    migrate_setting(MapPortNatpmp, "fUseNatpmp", "natpmp");
+    migrate_setting(Listen, "fListen", "listen");
+    migrate_setting(Server, "server", "server");
```

Comment:
Is this needed, given that no version was released with this setting?

Reply:
> Is this needed, given that no version was released with this setting?

Not strictly needed, but it would be strange to remove this because removing it would treat this setting differently than all the other settings, it causes no harm, it is one line of code, removing it would introduce strange behaviors trying to revert and test this PR, and removing it would introduce a bug in the PR if there is any release before this PR is merged.